UE/Tests/Application/Ports/BtsPortTestSuite.cpp

```diff
@@ -11,203 +11,212 @@
+                                    common::PhoneNumber{},
                                     PHONE_NUMBER};
-    message.writeText("Ala ma kota");
-    messageCallback(message.getMessage());
-}
+        msg.writeNumber(false);
+        messageCallback(msg.getMessage());
+    }
+
+    TEST_F(BtsPortTestSuite, shallSendAttachRequest)
+    {
+        common::BinaryMessage msg;
+        EXPECT_CALL(transportMock, sendMessage(_)).WillOnce(SaveArg<0>(&msg));
+        objectUnderTest.sendAttachRequest(BTS_ID);
+        common::IncomingMessage reader(msg);
+        ASSERT_NO_THROW(EXPECT_EQ(common::MessageId::AttachRequest, reader.readMessageId()));
+        ASSERT_NO_THROW(EXPECT_EQ(PHONE_NUMBER, reader.readPhoneNumber()));
+        ASSERT_NO_THROW(EXPECT_EQ(common::PhoneNumber{}, reader.readPhoneNumber()));
+        ASSERT_NO_THROW(EXPECT_EQ(BTS_ID, reader.readBtsId()));
+        ASSERT_NO_THROW(reader.checkEndOfMessage());
+    }
+
+    TEST_F(BtsPortTestSuite, shallHandleReceivedSms)
+    {
+        EXPECT_CALL(handlerMock, handleSmsReceived);
+        common::OutgoingMessage message{common::MessageId::Sms,
+                                        common::PhoneNumber{},
+                                        PHONE_NUMBER};
+        messageCallback(message.getMessage());
+    }
+
+    TEST_F(BtsPortTestSuite, shallSendMsg)
+    {
+        common::BinaryMessage msg;
+        EXPECT_CALL(transportMock, sendMessage(_)).WillOnce(SaveArg<0>(&msg));
+        objectUnderTest.sendMsg(common::PhoneNumber{115}, "test");
+        common::IncomingMessage reader(msg);
+        ASSERT_NO_THROW(EXPECT_EQ(common::MessageId::Sms, reader.readMessageId()));
+        ASSERT_NO_THROW(EXPECT_EQ(PHONE_NUMBER, reader.readPhoneNumber()));
+        ASSERT_NO_THROW(EXPECT_EQ(common::PhoneNumber{115}, reader.readPhoneNumber()));
+        ASSERT_NO_THROW(EXPECT_EQ("test", reader.readRemainingText()));
+        ASSERT_NO_THROW(reader.checkEndOfMessage());
+    }
+
+    TEST_F(BtsPortTestSuite, shallHandleCallRequest)
+    {
+        auto phoneNumber = common::PhoneNumber{100};
+        EXPECT_CALL(handlerMock, handleCallRequest(phoneNumber));
+        common::OutgoingMessage message{common::MessageId::CallRequest,
+                                        phoneNumber,
+                                        PHONE_NUMBER};
+        messageCallback(message.getMessage());
+    }
+
+    TEST_F(BtsPortTestSuite, shallHandleCallAccept)
+    {
+        auto phoneNumber = common::PhoneNumber{100};
+        EXPECT_CALL(handlerMock, handleCallAccepted(phoneNumber));
+        common::OutgoingMessage message{common::MessageId::CallAccepted,
+                                        common::PhoneNumber{100},
+                                        PHONE_NUMBER};
+        messageCallback(message.getMessage());
+    }
+
+    TEST_F(BtsPortTestSuite, shallHandlePeerUeBecomesUnknownCallAccepted)
+    {
+        EXPECT_CALL(handlerMock, handlePeerUeBecomesUnknown());
```

Comment:
Przez to testy nie przechodzą. Tzn. do żadnego z tych ifów program nie wchodzi :(
Spróbuj w OutgoingMessage ustawić jakieś pole z headerem starej wiadomości, wystarczy w sumie samo jej ID jakoś dodać.